javaagent-tooling/src/test/java/io/opentelemetry/javaagent/tooling/PropagatorsInitializerTest.java

```diff
@@ -0,0 +1,156 @@
+  }
+
+  @Test
+  void initialize_preconfiguredSameAsId() {
+    List<String> ids = singletonList("jaeger");
+    AtomicReference<ContextPropagators> seen = new AtomicReference<>();
+    Consumer<ContextPropagators> setter = seen::set;
+    Supplier<TextMapPropagator> preconfigured = () -> PropagatorsInitializer.Propagator.JAEGER;
+
+    PropagatorsInitializer.initializePropagators(ids, preconfigured, setter);
+
+    assertThat(seen.get().getTextMapPropagator())
+        .isSameAs(PropagatorsInitializer.Propagator.JAEGER);
+  }
+
+  @Test
+  void initialize_preconfiguredDuplicatedInIds() {
+    List<String> ids = Arrays.asList("b3", "jaeger", "b3");
+    AtomicReference<ContextPropagators> seen = new AtomicReference<>();
+    Consumer<ContextPropagators> setter = seen::set;
+    Supplier<TextMapPropagator> preconfigured = () -> PropagatorsInitializer.Propagator.JAEGER;
+
+    PropagatorsInitializer.initializePropagators(ids, preconfigured, setter);
+
+    assertThat(seen.get().getTextMapPropagator())
+        .extracting("textPropagators")
+        .isInstanceOfSatisfying(
+            TextMapPropagator[].class,
+            p ->
+                assertThat(p)
+                    .containsExactlyInAnyOrder(
+                        PropagatorsInitializer.Propagator.B3,
+                        PropagatorsInitializer.Propagator.JAEGER));
+  }
+
+  @Test
+  void initialize_justOneId() {
+    List<String> ids = singletonList("jaeger");
+    AtomicReference<ContextPropagators> seen = new AtomicReference<>();
+    Consumer<ContextPropagators> setter = seen::set;
+    Supplier<TextMapPropagator> preconfigured = TextMapPropagator::noop;
+
+    PropagatorsInitializer.initializePropagators(ids, preconfigured, setter);
+
+    assertThat(seen.get().getTextMapPropagator())
+        .isSameAs(PropagatorsInitializer.Propagator.JAEGER);
+  }
+
+  @Test
+  void initialize_idsWithNoPreconfigured() {
+    List<String> ids = Arrays.asList("b3", "unknown-but-no-harm-done", "jaeger");
+    AtomicReference<ContextPropagators> seen = new AtomicReference<>();
+    Consumer<ContextPropagators> setter = seen::set;
+    Supplier<TextMapPropagator> preconfigured = TextMapPropagator::noop;
+
+    PropagatorsInitializer.initializePropagators(ids, preconfigured, setter);
+
+    assertThat(seen.get().getTextMapPropagator())
+        .extracting("textPropagators")
+        .isInstanceOfSatisfying(
+            TextMapPropagator[].class,
+            p ->
+                assertThat(p)
+                    .containsExactlyInAnyOrder(
+                        PropagatorsInitializer.Propagator.B3,
+                        PropagatorsInitializer.Propagator.JAEGER));
+  }
+
+  @Test
+  void initialize_idsAndPreconfigured() {
+    List<String> ids = Arrays.asList("jaeger", "xray");
+    AtomicReference<ContextPropagators> seen = new AtomicReference<>();
+    Consumer<ContextPropagators> setter = seen::set;
+    TextMapPropagator mockPreconfigured = mock(TextMapPropagator.class);
```

Comment:
Just a minor suggestion: you can pull `seen` and `mockPreconfigured` to the class level, since they're used in almost all test cases